src/database/core/util/util.ts

```diff
@@ -416,15 +401,17 @@ export const ObjectToUniqueKey = function(obj) {
  * @param {!number} segsize The maximum number of chars in the string.
  * @return {Array.<string>} The string, split into appropriately-sized chunks
  */
-export const splitStringBySize = function(str, segsize) {
-  if (str.length <= segsize) {
+export const splitStringBySize = function (str: string, segsize: number): string[] {
+  const len = str.length;
+
+  if (len <= segsize) {
     return [str];
   }
 
-  var dataSegs = [];
-  for (var c = 0; c < str.length; c += segsize) {
-    if (c + segsize > str) {
```

Comment:
This condition would never be true since it's comparing a number with a string. It was probably meant to say `if (c + segsize > str.length) {`